docs/source/basics/tokenizers.rst

```diff
@@ -0,0 +1,320 @@
+        "user": ("User: ", "\\n"),
+        "assistant": ("Assistant: ", "\\n"),
+        "ipython": ("Tool: ", "\\n"),
+    }
+    m_tokenizer = mistral_tokenizer(
+        path="/tmp/Mistral-7B-v0.1/tokenizer.model"
+        prompt_template=template,
+    )
+
+.. code-block:: yaml
+
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      prompt_template:
+        system:
+          - "System: "
+          - "\\n"
+        user:
+          - "User: "
+          - "\\n"
+        assistant:
+          - "Assistant: "
+          - "\\n"
+        ipython:
+          - "Tool: "
+          - "\\n"
+
+If you don't want to add a prepend/append tag to a role, you can just pass in an empty string "" where needed.
+
+For more advanced customization of prompt templates, see :ref:`prompt_templates_usage_label`.
+
+.. TODO (RdoubleA) add a section on how to define prompt templates for inference once generate scsript is finalized
+
+Special tokens
+--------------
+
+Special tokens are model-specific tags that are required to prompt the model. They are different from prompt templates
+because they are assigned their own unique token IDs. For an extended discussion on the difference between special tokens
+and prompt templates, see :ref:`prompt_templates_usage_label`.
+
+Special tokens are automatically added to your data by the model tokenizer and do not require any additional configuration
+by the user. You also have the ability to customize the special tokens for experimentation by passing in a file path to
+the new special tokens mapping in a JSON file. This will NOT modify the underlying ``tokenizer.model`` to support the new
+special token ids - it is the user's responsibility to ensure that the tokenizer file encodes it correctly. Note also that
+some models require the presence of certain special tokens for proper usage.
+
+For example, here we change the ``"<|begin_of_text|>"`` and ``"<|end_of_text|>"`` token IDs in Llama3:
+
+.. code-block:: python
+
+    # special_tokens.json
+    {
+        "added_tokens": [
+            {
+                "id": 128257,
+                "content": "<|begin_of_text|>",
+            },
+            {
+                "id": 128258,
+                "content": "<|end_of_text|>",
+            },
+            # Remaining required special tokens
+            ...
+        ]
+    }
+
+.. code-block:: python
+
+    from torchtune.models.llama3 import llama3_tokenizer
+
+    tokenizer = llama3_tokenizer(
+        path="/tmp/Meta-Llama-3-8B-Instruct/original/tokenizer.model",
+        special_tokens_path="special_tokens.json",
+    )
+    print(tokenizer.special_tokens)
+    # {'<|begin_of_text|>': 128257, '<|end_of_text|>': 128258, ...}
+
+Base tokenizers
+---------------
+
+:class:`~torchtune.modules.tokenizers.BaseTokenizer` are the underlying byte-pair encoding modules that perform the actual raw string to token ID conversion and back.
```

Comment:
```suggestion
:class:`~torchtune.modules.tokenizers.BaseTokenizer`s are the underlying byte-pair encoding modules that perform the actual raw string to token ID conversion and back.
```

Reply:
I think it renders incorrectly if I make it plural